### PR TITLE
Add support for /v0.6/stats

### PR DESCRIPTION
**Pull request by Kyle-Verhoog**
Note that support for snapshotting will be included in a follow-up
patch.

Dependent on #50

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -229,6 +229,26 @@ name where the snap will be stored and retrieved.
...
+
+#### [optional] `?test_session_token=`
+#### [optional] `X-Datadog-Test-Session-Token`
+
+Returns the requests in the following json format:
+
+```json
+[
+  {
+    "headers": {},
```
Comment (brettlangdon) [inline README.md:244]: is url + method included as well? should it be?

File: README.md [hunk 2]
```diff
@@ -237,6 +257,14 @@ Return traces that have been received by the agent for the given session token.
 #### [optional] `X-Datadog-Test-Session-Token`
 
 
+### /test/session/stats
+
+Return stats that have been received by the agent for the given session token.
+
+#### [optional] `?test_session_token=`
+#### [optional] `X-Datadog-Test-Session-Token`
```
Comment (brettlangdon) [inline README.md:265]: document the response body?

File: releasenotes/notes/trace-stats-97da3a478dabfa99.yaml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+---
+features:
+  - |
+    Add support for the /v0.6/stats endpoint. All requests made to the endpoint
```
Comment (brettlangdon) [inline releasenotes/notes/trace-stats-97da3a478dabfa99.yaml:4]:
```suggestion
    Add support for the ``/v0.6/stats`` endpoint. All requests made to the endpoint
```

?

File: tests/test_session.py [hunk 4]
```diff
@@ -108,3 +111,30 @@ async def test_two_sessions(
...
+    assert resp.status == 200, await resp.text()
+    resp = await do_reference_v06_http_stats(token="test_case")
+    assert resp.status == 200, await resp.text()
+
+    resp = await agent.get(
+        "/test/session/requests", params={"test_session_token": "test_case"}
+    )
+    requests = await resp.json()
+    assert resp.status == 200
+    assert len(requests) == 3
```
Comment (brettlangdon) [inline tests/test_session.py:135]: should we verify that we have 2 v04 and 1 v06 request here?

File: ddapm_test_agent/agent.py [hunk 5]
```diff
@@ -309,13 +336,19 @@ async def handle_session_traces(self, request: Request) -> web.Response:
...
+    async def handle_session_tracestats(self, request: Request) -> web.Response:
+        token = request["session_token"]
+        stats = await self._tracestats_by_session(token)
+        return web.json_response(stats)
+
     async def handle_session_requests(self, request: Request) -> web.Response:
         token = request["session_token"]
         resp = []
-        for req in self._requests_by_session(token):
+        for req in reversed(self._requests_by_session(token)):
```
Comment (brettlangdon) [inline ddapm_test_agent/agent.py:347]: why this change?

**Review by brettlangdon — APPROVED** [review]

File: ddapm_test_agent/agent.py (hunk 5 above)
Comment (Kyle-Verhoog) [reply ddapm_test_agent/agent.py:347]: Yeah, should have been included in #50, or earlier in `_requests_by_session`. `_requests_by_session` returns the requests in reverse order of being received.

So this is just putting them in order of being received which makes more sense to me.

I'll put up a patch following this one to reverse the order of `_requests_by_session`
